libchisel/src/instructionerrors.rs

```diff
@@ -0,0 +1,39 @@
+use parity_wasm::elements::Instruction;
+use std::error;
+use std::fmt;
+
+#[derive(Debug)]
+pub enum InstructionError {
+    GlobalNotFound,
+    LocalNotFound,
+    UnmatchedInstruction,
+    InvalidOperation(Instruction),
+}
+
+impl fmt::Display for InstructionError {
```

Comment:
If we merge `ModuleError` and `InstructionError`, could you create a new PR which just adds the `Display` and `Error` traits on `ModuleError`? Then this PR can just extend that with the new enum values.

Reply:
Alternatively, instead of adding new variants, it may make sense to just encode these errors as a string in the `Custom` variant of `ModuleError`.

Reply:
Or make `Custom(T)` type generic and throw `InstructionError` in there.

Reply:
Text in general is fine, but for programmatic detection the enums are better.

Reply:
I'd like to stick to enum variants if possible, and I can certainly look into merging `InstructionError` into `ModuleError`.